sdk/iOS/src/MSUser.h

```diff
@@ -40,7 +40,7 @@
 /// requests made to the Microsoft Azure Mobile Service, allowing the client to
 /// perform all actions on the Microsoft Azure Mobile Service that require
 /// authenticated-user level permissions.
-@property (nonatomic, copy)         NSString *mobileServiceAuthenticationToken;
+@property (nonatomic, copy, nonnull)         NSString *mobileServiceAuthenticationToken;
```

Comment:
User doesn't have a constructor taking both?  The client example shows code as so:

  self.todoService.client.currentUser = [[MSUser alloc] initWithUserId:userid];
  self.todoService.client.currentUser.mobileServiceAuthenticationToken = [SSKeychain passwordForService:@"AzureMobileServiceTutorial" account:userid];

So its possible for the token to be null in a valid flow